### PR TITLE
ab micro: update on experimental setup change

### LINKED ISSUE #691: AB Micro: Update Microscopy 'Analyze' page when Experimental Setup is changed.

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?

Fix #691 
#### What's this PR do?

Keeps the MicroscopyImageMappings up to date with the changes in the Experimental Setup:
1 - 'Select Variable' changes 
2 - add/delete 'Strains'
3 - add/delete 'Treatments', 'Temperature', 'Collection Time'
4 - enable/disable 'StrainProtocols'

### CODE REVIEW

**Comment by pdpinch** [comment]
I'm not exactly sure how to test this, but I tried creating a microscopy experiment with only one treatment and uploaded an image for it. Then I added another treatment, and more variables. 

The new treatment was available in the Microscopy Analyze form, but the image I uploaded was gone (which I expected) but there was no way to upload any more images to the original treatment. Here's a screenshot:

screen shot 2016-08-21 at 9 06 54 pm

This seems like a bug to me, but I'm not 100% sure.

**Comment by annagav** [comment]
I am not able to reproduce this. I would need to see the right order of steps when you do this.

**Comment by pdpinch** [comment]
I think it was a migration issue. I destroyed the database and started from scratch, and it seems to work fine now.

**Comment by annagav** [comment]
@pdpinch Is this good for merge, or still needs code review?

**Comment by pdpinch** [comment]
This needs a rebase.

**Comment by annagav** [comment]
Rebased

File: instructor/common.py [hunk 1]
```diff
@@ -452,15 +452,7 @@ def assignments_variables(request):
...
             if form.has_changed():
-                models.Treatment.objects.filter(assignment=assignment).delete()
-                models.StrainTreatment.objects.filter(
-                    assignment=assignment
-                ).delete()
-                models.WesternBlotBands.objects.filter(
-                    antibody__western_blot__assignment=assignment
-                ).delete()
-                create_treatments(assignment)
-                update_wb_bands(assignment)
```
Comment (blarghmatey) [inline instructor/common.py:463]: Was the removal of this method call intentional?

**Comment by blarghmatey** [comment]
Other than my one comment this looks good to go 👍

File: instructor/common.py (hunk 1 above)
Comment (annagav) [reply instructor/common.py:463]: Yes, it is being called already within `create_treatments`...

**Pull request merged**